### PR TITLE
Implement ElasticSearch

### PR DESCRIPTION
**Pull request by david-mears-dfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-20210401

## Changes in this PR:

- Implement ElasticSearch

### CODE REVIEW

File: app/services/search/strategies/elasticsearch.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def total_count
+    return 0 if vacancies.none?
+
+    vacancies.raw_answer["nbHits"]
+  end
+
+  private
+
+  def big_if_true(bool)
+    bool ? "big" : "small"
```
Comment (ddippolito) [inline app/services/search/strategies/elasticsearch.rb:34]: 🚫 🙅 

Are you sure this is correct? 
What's big? What's small?
Can you extract this in well defined global constants?

Comment (ddippolito) [reply app/services/search/strategies/elasticsearch.rb:34]: @DFE-Digital/teaching-vacancies-developers discuss!

Comment (david-mears-dfe) [reply app/services/search/strategies/elasticsearch.rb:34]: Happy to use global variables if you prefer?

**Comment by david-mears-dfe** [comment]
This was an April Fool's joke in case anyone was confused.

**Pull request closed without merging**